### PR TITLE
LUCENE-9639: Add unit tests for SimpleTextVector format

### PR DESCRIPTION
**Pull request by zacharymorn**
# Description

Added unit tests for SimpleTextVector format, and made a few changes in SimpleTextVectorReader and SimpleTextVectorWriter to pass the tests 

# Solution

Added unit tests for SimpleTextVector format by extending BaseVectorFormatTestCase

# Tests

* Added unit tests for SimpleTextVector format by extending BaseVectorFormatTestCase
* Tests passed but the nocommit comment is currently failing pre-commit

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ private static IndexInput openDataInput(
 
     String fileName =
         IndexFileNames.segmentFileName(state.segmentInfo.name, state.segmentSuffix, fileExtension);
+    // nocommit should the openInput call be put in try-finally as well for IOException handling?
```
Comment (msokolov) [inline lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java:123]: I think it's OK - the exception handling got moved to the outer scope in the PR I referenced above

**Review by msokolov — COMMENTED** [review]
Thanks for the PR! I just have one comment about exception handling, otherwise looks good

File: lucene/codecs/src/java/org/apache/lucene/codecs/simpletext/SimpleTextVectorReader.java [hunk 2]
```diff
@@ -93,22 +94,35 @@
             readState.segmentInfo.name,
             readState.segmentSuffix,
             SimpleTextVectorFormat.VECTOR_EXTENSION);
-    dataIn = readState.directory.openInput(vectorFileName, IOContext.DEFAULT);
+
+    boolean success = false;
```
[the remark this answers is not in the document]
Comment (zacharymorn) [reply lucene/codecs/src/java/org/apache/lucene/codecs/simpletext/SimpleTextVectorReader.java]: Makes sense. I've extended the try-finally block to contain meta file input as well.

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java (hunk 1 above)
Comment (zacharymorn) [reply lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java]: Sounds good. I have removed the nocommit.

**Comment by zacharymorn** [comment]
> Thanks for the PR! I just have one comment about exception handling, otherwise looks good

Thanks Michael for the review! I've pushed an update to improve on the exception handling like you suggested. Please let me know if it looks good to you.

**Review by msokolov — APPROVED** [review]
Thanks, @zacharymorn. I'll push this now

**Pull request merged**